pyproject.toml

```diff
@@ -36,48 +36,50 @@ exclude = ["tests", "tests.*"]
 hathor-cli = 'hathor.cli.main:main'
 
 [tool.poetry.dev-dependencies]
-flake8 = "~3.8.4"
-isort = {version = "~5.7.0", extras = ["colors"]}
-yapf = "^0.30"
-mypy = "^0.790"
-pytest = "~6.2.1"
-pytest-cov = "~2.11.0"
+flake8 = "~4.0.1"
+isort = {version = "~5.10.1", extras = ["colors"]}
+yapf = "^0.32"
+mypy = {version = "^0.931", markers = "implementation_name == 'cpython'"}
+mypy-zope = {version = "^0.3", markers = "implementation_name == 'cpython'"}
+pytest = "~6.2.5"
+pytest-cov = "~3.0.0"
 flaky = "~3.7.0"
+# stubs:
+types-requests = "~2.27.7"
+types-pyopenssl = "~21.0.3"
 
 [tool.poetry.dependencies]
 python = ">=3.7,<4"
-twisted = "~20.3.0"
-autobahn = "~20.12.3"
-base58 = "~2.1.0"
+twisted = "~21.7.0"
+autobahn = "~21.11.1"
+base58 = "~2.1.1"
 colorama = "~0.4.4"
-configargparse = "~1.2.3"
-cryptography = "~3.3.1"
-graphviz = "~0.16"
-ipython = "<7.17"  # 7.17 drops Python 3.6 compatibility
-jedi = "<0.18"  # indirect dependency restriction (0.18 breaks ipython)
-mnemonic = "~0.19"
-prometheus_client = "~0.9.0"
-pyopenssl = "~20.0.1"
+configargparse = "~1.5.3"
+cryptography = "~36.0.1"
+graphviz = "~0.19.1"
+ipython = "<8"  # ipython 8.0 drops compatibility with Python 3.7
```

Comment:
Should we drop compatibility with Python 3.7 as well? I'm currently using 3.7.10 in my dev environment.

Reply:
At the very least I'd wait a few releases. It is still the default version of the docker image when no version is given, which means is what we use by default on our deploys. Also, but much less relevant, the latest version of PyPy is only compatible with Python-3.7 and Python-3.8, but unfortunately PyPy-Python-3.8 cannot build python-rocksdb, it has something to do with Cython, which may be updated in the future to support that version combination.

Reply:
Good points. Let's keep it until everything works fine with 3.8+. Thanks!